### PR TITLE
fix dashboard entity recursive param

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #15137

### CODE REVIEW

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Review by cedric-anne — COMMENTED** [review]
Maybe another solution is to use a unique `json_encode` on all params, to ensure that JS will not be malformed.

```php
$js_params = json_encode([
    'current' => $this->current,
    'cols'      => $this->cols,
    ...
    'is_recursive' => $_SESSION['glpiactive_entity_recursive'] ? 1 : 0
]);
$js = <<<JAVASCRIPT
    $(function () {
        new GLPIDashboard({$js_params});
      });
JAVASCRIPT;
```

**Comment by cconard96** [comment]
> Maybe another solution is to use a unique `json_encode` on all params, to ensure that JS will not be malformed.
> 
> ```
> $js_params = json_encode([
>     'current' => $this->current,
>     'cols'      => $this->cols,
>     ...
>     'is_recursive' => $_SESSION['glpiactive_entity_recursive'] ? 1 : 0
> ]);
> $js = <<<JAVASCRIPT
>     $(function () {
>         new GLPIDashboard({$js_params});
>       });
> JAVASCRIPT;
> ```

Doesn't seem possible without some other changes. A null token becomes "null" instead of just null.
Maybe its better to just move this to Twig in a separate PR and leave this as a quick/easy fix that people can apply.

**Comment by cconard96** [comment]
Done. I don't know if it broke it in some other cases though.

File: src/Dashboard/Grid.php [hunk 1]
```diff
@@ -413,24 +411,25 @@ public function show(bool $mini = false, ?string $token = null)
...
         $cache_key  = sha1($_SESSION['glpiactiveentities_string '] ?? "");
 
+        $js_params = json_encode([
+            'current'       => $this->current,
+            'cols'          => $this->grid_cols,
+            'rows'          => $this->grid_rows,
+            'cell_margin'   => $this->cell_margin,
+            'rand'          => $rand,
+            'ajax_cards'    => $ajax_cards,
+            'all_cards'     => $cards_json,
```
Comment (cedric-anne) [inline src/Dashboard/Grid.php:421]:
```suggestion
            'all_cards'     => $cards,
```

File: src/Dashboard/Grid.php [hunk 2]
```diff
@@ -413,24 +411,25 @@ public function show(bool $mini = false, ?string $token = null)
...
+            'all_widgets'   => $all_widgets_json,
```
Comment (cedric-anne) [inline src/Dashboard/Grid.php:422]:
```suggestion
            'all_widgets'   => $all_widgets,
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**